### PR TITLE
ADEN-5312 AppNexusAst - video outstream adapter

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/lookup/prebid/adaptersRegistry.js [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
+	'ext.wikia.adEngine.lookup.prebid.adapters.appnexusAst',
 	'ext.wikia.adEngine.lookup.prebid.adapters.audienceNetwork',
 	'ext.wikia.adEngine.lookup.prebid.adapters.indexExchange',
 	'ext.wikia.adEngine.lookup.prebid.adapters.openx',
 	'ext.wikia.adEngine.lookup.prebid.adapters.rubicon',
 	'ext.wikia.adEngine.lookup.prebid.adapters.wikia',
 	'ext.wikia.adEngine.lookup.prebid.adapters.veles',
 	'wikia.window'
-], function(aol, appnexus, audienceNetwork, indexExchange, openx, rubicon, wikia, veles, win) {
+], function(aol, appnexus, appnexusAst, audienceNetwork, indexExchange, openx, rubicon, wikia, veles, win) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/lookup/prebid/adaptersRegistry.js]: MAJOR Function has 10 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/appnexusAst.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	var bidderName = 'appnexusAst',
+		debugPlacementId = '5768085',
+		slots = {
+			oasis: {
+				TOP_LEADERBOARD: {
+					placementId: '5768085' // FIXME
+				},
+				INCONTENT_PLAYER: {
+					placementId: '5768085' // FIXME
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/lookup/prebid/adapters/appnexusAst.js]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/appnexusAst.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+				}
+			},
+			mercury: {
+				MOBILE_IN_CONTENT: {
+					placementId: '5768085' // FIXME
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/lookup/prebid/adapters/appnexusAst.js]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/appnexusAst.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	'wikia.location'
+], function (slotsContext, geo, instantGlobals, loc) {
+	'use strict';
+
+	var bidderName = 'appnexusAst',
+		debugPlacementId = '5768085',
+		slots = {
+			oasis: {
+				TOP_LEADERBOARD: {
+					placementId: '5768085' // FIXME
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/lookup/prebid/adapters/appnexusAst.js]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 14 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L183](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L183): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L208](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L208): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L208](https://github.com/Wikia/app/blob/7e945da11e45ea1803502dc7fdd2269f7ccae200/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L208): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 12 major
* ![INFO][INFO] 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L183](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L183): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L208](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L208): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L208](https://github.com/Wikia/app/blob/6f730f3757fd1490ebc50dcb1f14b4ad3b38aa76/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L208): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 12 major
* ![INFO][INFO] 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L183](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L183): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L208](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L208): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L208](https://github.com/Wikia/app/blob/b5290073a395673c3bbe0f6acd406766f5eddf2a/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L208): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 16 issues
* ![MAJOR][MAJOR] 15 major
* ![INFO][INFO] 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L183](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L183): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L208](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L208): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L208](https://github.com/Wikia/app/blob/c00c3c6465903189af902d8df5f822a9ec9b2f33/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L208): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 16 issues
* ![MAJOR][MAJOR] 15 major
* ![INFO][INFO] 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L96](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L96): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L97](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L97): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L176](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L176): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L201](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L201): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L201](https://github.com/Wikia/app/blob/8124614a349c6d86d71d2b534f3a2c56f991868d/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L201): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/appnexusAst.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+				}
+			}
+		};
+
+	function isEnabled() {
+		return geo.isProperGeo(instantGlobals.wgAdDriverAppNexusAstBidderCountries);
+	}
+
+	function prepareAdUnit(slotName, config) {
+		var isDebugMode = loc.href.search('appnexusast_debug_mode=1') >= 0;
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/lookup/prebid/adapters/appnexusAst.js]: What do you think of using `indexOf()` instead of `search()` here?

**Review by nandy-andy — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 16 issues
* ![MAJOR][MAJOR] 15 major
* ![INFO][INFO] 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L19](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L19): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L96](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L96): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L97](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L97): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L176](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L176): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L201](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L201): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L201](https://github.com/Wikia/app/blob/bb4283a792a8769ad18d8574997c493fe10976be/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L201): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**